### PR TITLE
mkdocs.yml,docs/dev-proc: initial version of sections describing dev …

### PR DESCRIPTION
**Pull request by pietrushnic**
…proc

Signed-off-by: Piotr Król <piotr.krol@3mdeb.com>

### CODE REVIEW

**Review by krystian-hebel — CHANGES_REQUESTED** [review]
I think we should also explicitly say that force-pushes are forbidden except:
- rebasing - when some other PR is merged to target branch before our does, or when upstream's master introduces the same fixes that our branch would (this has happened for CBFS endianness issues),
- squashing - to not produce unnecessary "fix indentation" or "add missing braces" commits to the history,
- (re-)signing commits (both `-S` and `-s`) - shouldn't happen, but if it does happen it would be better to have it fixed by original author than the person that tries to upstream it some time later.

All of the above reduce upstreaming effort mostly. Of course, force-pushes to anything else than `rel_vX.Y.Z` or `<feature>` branches should be forbidden unconditionally.

File: docs/dev-proc/standard-release-process.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+and should be maintained in that way.
+
+## Process steps
+
+1. Checkout new branch `<platform>/rel_x.y.z` from recent commit on
+   `<platfrom>/release` - to understand versioning scheme please read
+   [Versioning](versioning.md) section
+2. Merge current `<platform>/develop` to `<platform>/rel_x.y.z`
+3. Run platform regression test suite
+5. Fix all required issues and repeat point 4 until fixed - this doesn't mean
```
[the remark this answers is not in the document]
Comment (pietrushnic) [reply docs/dev-proc/standard-release-process.md]: Fixed

File: docs/dev-proc/feature-bug-fix-dev-process.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+1. Create automated test that validate feature or reproduce bug - test fails at
+   this point
+2. Pull upstream `master` branch to Dasharo forked repository `master` branch
+3. Merge `master` to `<platform>/develop`
+4. Create new branch `<platform>/<feature>` from `develop`
+5. Commit changes to `<platform>/<feature>`
+6. Run test written in point 1 and make sure it pass.
+7. Run `<platform>` regression test suite and make sure new feature does not
+   introduce new bugs.
+8. Submit PR to `develop`
```
[the remark this answers is not in the document]
Comment (pietrushnic) [reply docs/dev-proc/feature-bug-fix-dev-process.md]: Fixed

**Comment by pietrushnic** [comment]
@krystian-hebel I added information about force-pushes to source code structure.

File: docs/dev-proc/source-code-structure.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## New platform support
+
+Branch for new platform should be created from most recent`master` branch tag.
+If there is justified need to create support for new board at arbitrary
+non-tagged commit developer should mark this commit with `<platform>_v0.0.0`
+tag.
+
+## Force-pushes rules
+
+Force-pushes to `<platform>/rel_vX.Y.Z` or `<platfrom>/<feature>` are
```
Comment (krystian-hebel) [inline docs/dev-proc/source-code-structure.md:31]: Shouldn't this be the other way around? Other branches include `<platform>/develop` and `<platform>/release`, those shouldn't be force-pushed.

Also a typo in second `<platfROm>`

File: docs/dev-proc/standard-release-process.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+and should be maintained in that way.
+
+## Process steps
+
+1. Checkout new branch `<platform>/rel_vX.Y.Z` from recent commit on
+   `<platfrom>/release` - to understand versioning scheme please read
+   [Versioning](versioning.md) section
+2. Merge current `<platform>/develop` to `<platform>/rel_vX.Y.Z`
+3. Run platform regression test suite
+4. Fix all required issues and repeat point 4 until fixed - this doesn't mean
```
Comment (krystian-hebel) [inline docs/dev-proc/standard-release-process.md:17]:
```suggestion
4. Fix all required issues and repeat point 3 until fixed - this doesn't mean
```

File: docs/dev-proc/source-code-structure.md (hunk 3 above)
Comment (pietrushnic) [reply docs/dev-proc/source-code-structure.md:31]: Fixed

**Review by krystian-hebel — APPROVED** [review]

**Pull request merged**